### ISSUE TITLE
Release notes for 0.226

### ISSUE THREAD

**Issue by viczhang861** [prestodb/presto#13340]
Issue to track release notes for next release.

**Comment by neeradsomanchi** [comment]
## Andrii Rosa          
- [ ] All Checked
-  08-26 10:34:48    9fe9f13cb  Do not check bucketing consistency for unpartitioned tables
-  09-04 14:22:59    820a0b098  Update scm to point to facebookexternal
-  09-04 14:25:31    b4e09626b  Change FunctionMetadata to use FullyQualifiedName
-  09-04 15:47:48    1c04f9d64  Update esri version to 2.2.3
-  09-04 15:47:48    3e803381e  Do not use tempfs in TestPrismIntegrationSmokeTest
-  09-04 15:47:48    75b0169de  Increase task writer count for PrismQueryRunner
## James Sun            
- [ ] All Checked
-  09-03 11:50:09    b521db68f  Filter Pushdown with String Functions
-  09-03 11:50:51    0fd0b1b5f  Add Facebook Raptor query runner
## Rongrong Zhong       
- [ ] All Checked
-  08-29 11:04:30    774d0f5d6  Add UDFs FB_PHOTODNA_DISTSQ and FB_PHOTODNA_DISTSQ_LESS_THAN_OR_EQUAL
-  09-05 16:37:06    d967fdd09  Catch up with changes to HiveTableLayoutHandle's constructor
## Shixuan Fan          
- [ ] All Checked
-  08-21 11:56:12    30a4182e4  Write WarmStorageOperationEvent to scribe
-  08-21 11:56:12    e85978de6  Separate PrismFederationModule for Hive federation
-  08-26 11:02:11    ecc02832b  Bind WarmStorageEventClientConfig correctly with prefix
-  08-30 11:20:13    e8c1645e7  Make warm storage operation event scribe category configurable
-  09-05 11:11:25    d27743073  Fix rethrow logic in TimeoutDirectoryLister
## Valery Yundin        
- [ ] All Checked
-  09-05 11:35:57    a2da08d87  Implement FB_LOGGER_SAMPLING UDF
## Vic Zhang            
- [ ] All Checked
-  08-21 10:26:49    9138b935c  prepare for next development iteration - 0.226-SNAPSHOT
-  08-21 20:11:38    487cf37a2  Fix import in release script
-  09-05 15:28:09    8ab920101  Write query task count to presto_query_execution_stats table

**Comment by neeradsomanchi** [comment]
## Andrii Rosa          
- [ ] All Checked
-  08-23 12:00:01    b508153ba7 Update fixed-precision docs to reflect constants parsed as DECIMAL
-  08-29 17:11:57    5d18a1c01d Check Parquet schema mismatch
-  08-30 07:50:44    d591d7970b Allow anonymous rows in Hive temporary tables
-  08-30 09:30:49    139e024327 Remove StatisticAggregationsDescriptor from TableWriterNode
-  08-30 09:30:57    88fe29d2e3 Remove unnecessary row expression to symbol expression translation
-  08-30 13:39:34    dd93677651 Use Optional instead of null when getting Parquet types
-  09-03 13:46:28    916805a2b9 Merge TableWriter output before sending to TableFinishOperator
-  09-03 13:46:28    f0009dca3c Refactor TableFinishOperator and TableWriterOperator
-  09-03 19:50:12    c6371ab702 Fix grouped execution tagging for table scan
-  09-04 22:11:01    2de28d2493 Add query task count to statistics field of QueryCompletedEvent
-  09-05 16:31:23    0244ccd9c9 Allow recoverable execution for parallel writes
-  09-05 16:31:23    59a230a1f8 Allow local exchanges on non system partitioning
-  09-05 16:31:23    85efb9356a Refactor NodePartitioningManager
-  09-05 16:31:23    c594611edd Allow concurrent writes into a bucketed table
## James Sun            
- [ ] All Checked
-  08-22 15:39:09    0f2b3dd972 Move TranslateExpressions above last InlineProjections
-  08-22 15:39:09    8b5d746d49 Move TranslateExpressions above AddLocalExchanges
-  08-22 15:39:09    9a0824d8eb Move TranslateExpressions above last UnaliasSymbolReferences
-  08-22 15:39:09    cdec4f275b Move TranslateExpressions above second InlineProjections to last
-  08-22 15:39:09    dc11cdadcc Fix function name for try expression
-  08-22 15:39:09    ef724d293b Do not print unresolved plan in assertPlan
-  08-22 15:39:09    ffe309dd76 Move TranslateExpressions above last projectionPushDown
-  08-23 15:49:19    3fade37fad Increase jvm heap for mvn build
-  08-23 15:58:04    6e220a114e Cache row group column stats in ORC column writer
-  09-02 22:15:40    97b9c11bcc Fix ShardCleaner not running on HDFS
## Leiqing Cai          
- [ ] All Checked
-  08-22 23:42:55    f1bbb82267 Mark method as alwaysRun when closing TestingMySqlServer
## Maria Basmanova      
- [ ] All Checked
-  08-21 23:22:32    394c8c3f68 Add benchmarks for Flatbush and JTS STR RTree
-  08-21 23:22:32    5a4c520638 Use Flatbush instead of JTS STR RTree for spatial joins
-  08-21 23:22:32    879775e2fe Ignore order of rows in spatial join tests
-  08-21 23:22:32    89aff2b9f5 Add HasExtent interface
-  08-21 23:22:32    9124aa4783 Implement Flatbush Hilbert RTree
-  08-22 14:48:35    1bd45da521 Add FloatSelectiveStreamReader
-  08-22 16:08:15    8492bfa98b Update memory usage while building R-Tree
-  08-23 11:35:25    0a41532718 Avoid calling SelectiveStreamReader#read with zero positions to read
-  08-23 11:35:25    2af400e2cd Add StructSelectiveStreamReader
-  08-23 11:35:25    688e612a4b Add subfield pruning for structs
-  08-23 11:35:25    6d22ca98c8 Add ByteSelectiveStreamReader::toString
-  08-23 11:35:25    a9726d4916 Add more tests for ListSelectiveStreamReader
-  08-27 03:25:58    2d7191891a Adds testLength to TupleDomainFilter
-  08-27 08:03:54    09bc153b7c Move Arrays class from presto-orc to presto-array
-  08-28 03:06:52    d47b4e40c5 Remove unused OrcReader field maxReadSize
-  08-28 03:10:29    bd304e8a13 Fix docs for geospatial functions
-  08-28 04:05:20    47423bbb2c Remove unused systemMemoryContext from OrcBatchRecordReader
-  08-28 11:21:38    0ea8208235 Fix handling of non-deterministic filters in SelectiveStreamReaders
-  09-03 07:32:27    60a5a5c379 Reuse compressionBuffer when serializing pages
-  09-03 09:56:18    2d19b6d82b Allow special character in MongoDB password
-  09-03 09:56:18    70545f41bb Add test for MongoClientConfig
-  09-03 12:22:56    643ee08db0 Remove unused NULL_ENTRY_SIZE from OrcBatchPageSource
-  09-03 12:22:56    ea257f5c15 Simplify block creation for all null values
-  09-03 19:59:26    06057380fd Test for ST_Buffer NPE fix
-  09-03 19:59:26    3cdf6803a3 Fix centroids for small/degenerate polygons
-  09-03 19:59:26    dce764376a Upgrade ESRI geometry to v2.2.3
-  09-04 11:24:57    926ca277ac Add MapSelectedStreamReader
-  09-04 11:24:57    af15a75a87 Add positionCount parameter to createBlockFromKeyValue
-  09-04 11:24:57    e31ac556ee Add subfields used only for filtering to the list of required subfields
-  09-05 08:12:22    2e17628224 Add more tests for ListSelectiveStreamReader
-  09-05 08:12:22    6cffd16543 Add more checks to OrcTester.assertFileContentsPresto
-  09-05 08:12:22    7b9b0cef55 Add support for filters on subfields to OrcTester
-  09-05 09:39:41    7c28d0b0de Fix reading empty and all-nulls lists with filters
-  09-05 12:06:35    09eddceefc Expose formatRowExpression to the connectors
-  09-05 12:06:35    6c1aab7336 Add pushed down filter to the output of EXPLAIN
-  09-05 12:06:35    bca2bce65d Remove session parameter from RowExpressionFormatter's constructor
-  09-05 15:44:22    29bb4c60f8 Optimize single-column FilterFunctions on DictionaryBlocks
## Rebecca Schlussel    
- [ ] All Checked
-  08-26 07:26:32    6d296f9068 Make PrestoS3InputStream#read comply with read method contract on EOF
-  08-27 08:34:54    b2dd996520 Add .mailmap to increase release stats accuracy
-  08-30 09:16:20    cbe93751a2 Add test for PrestoS3InputStream positional read EOS
-  09-03 08:01:41    990545dbfd Add release note for 0.225
-  09-04 07:46:50    d0d8850143 Improve condition for ignoring bucketing for hive
-  09-05 13:03:56    67efdcf133 Move request slug from path to query
## Rongrong Zhong       
- [ ] All Checked
-  08-22 10:38:12    1fc8e5b6c5 Fix QueryContext.getAdditionalFailureInfo bug
-  08-23 08:42:08    0d7da7a8bd Disable TestMySqlCaseInsensitiveMapping
-  09-04 11:33:30    272cd9518d Move SqlFunction to spi
-  09-04 11:33:30    5149d92be7 Move function resolution logic to FunctionManager
-  09-04 11:33:30    85d30a6314 Move FunctionNamespaceManager to SPI
-  09-04 11:33:30    9622d26a8c Change Signature to use FullyQualifiedName
-  09-04 11:33:30    d8f34b0f56 Introduce FullyQualifiedName in SPI
-  09-04 11:33:30    e23005fa16 Break FunctionNamespaceManager.resolveFunction
-  09-04 11:33:30    e4daef33d0 Rename FunctionNamespace to FunctionNamespaceManager
-  09-04 11:33:30    fa36f4d917 Remove StaticFunctionNamespaceFactory
-  09-04 11:33:31    30528632e2 Rename StaticFunctionNamespace to BuiltInFunctionNamespace
-  09-04 11:33:31    3379b4f497 Resolve function using all registered function namespaces
-  09-04 11:33:31    c359c2196d Allow FunctionNamespace plugin
## Sahar Massachi       
- [ ] All Checked
-  08-29 11:21:35    95b2b3307d Add DoubleSelectiveStreamReader
## Vic Zhang            
- [ ] All Checked
-  08-21 10:25:55    f455e2019c prepare for next development iteration - 0.226-SNAPSHOT
## Wenlei Xie           
- [ ] All Checked
-  08-29 15:07:05    d960aef256 Improve error message when resource group selector cannot be loaded
-  09-04 18:13:48    ecdbbb3edc Add debug support for offline table and partition
-  09-04 20:50:09    523501d69b Inline LocalQueryRunner#translateExpressions
## Zhenxiao Luo         
- [ ] All Checked
-  08-31 08:07:29    1dfa608474 Fix Elasticsearch Connector scroll-timeout documentation

**Comment by mbasmanova** [comment]
```
General
--------
* Use Hilbert Packed RTree (Flatbush) instead of the JTS STR RTree for spatial joins.  The Flatbush is a flattened tree that uses vastly fewer heap allocations, and has faster build and query times.
```

**Comment by rongrong** [comment]
Not sure whether we want release notes from my changes. It adds some new APIs to SPI but it's not really ready for use. So I think we should skip release notes for now.

**Comment by neeradsomanchi** [comment]
# Missing Release Notes
## Andrii Rosa
- [ ] all checked
- https://github.com/wei-yi/presto/pull/1 get new code
- https://github.com/prestodb/presto/pull/12550 Refine Parquet schema mismatch message
- c594611edd836a9f454678e808ad56ff6986e65b Allow concurrent writes into a bucketed table
- 59a230a1f874a57e659099faa38c27766a8643b6 Allow local exchanges on non system partitioning
- 85efb9356acec97a7fae3bc8c8f67e91c351ef6d Refactor NodePartitioningManager
## James Sun
- [ ] all checked
- https://github.com/prestodb/presto/pull/13275 Increase jvm heap for mvn build
- https://github.com/prestodb/presto/pull/13047 Move row expression above last projectionPushDown
## Maria Basmanova
- [x] all checked
- https://github.com/prestodb/presto/pull/13289 Optimize single-column FilterFunctions to only run on distinct values for DictionaryBlocks
- https://github.com/prestodb/presto/pull/13064 Allow special character in MongoDB password
- https://github.com/prestodb/presto/pull/13278  Fix handling of non-deterministic filters in SelectiveStreamReaders
- https://github.com/prestodb/presto/pull/13286 Adds testLength to TupleDomainFilter
## Rongrong Zhong
- [x] all checked
- https://github.com/prestodb/presto/pull/12743 Function resolution
- https://github.com/prestodb/presto/pull/13268 Disable TestMySqlCaseInsensitiveMapping
- https://github.com/prestodb/presto/pull/13120 Fix QueryContext.getAdditionalFailureInfo bug
## Wenlei Xie
- [ ] all checked
- https://github.com/prestodb/presto/pull/13036 Improve error message when resource group selector cannot be loaded
## Zhenxiao Luo
- [ ] all checked
- https://github.com/prestodb/presto/pull/13257 elasticsearch.scroll-timeout example correction


# Extracted Release Notes
- #13079 Maria Basmanova: Use Flatbush RTree for Spatial Joins
  - Use Hilbert Packed RTree (Flatbush) instead of the JTS STR RTree for spatial joins.  The Flatbush is a flattened tree that uses vastly fewer heap allocations, and has faster build and query times.
- #13232 Maria Basmanova: Reuse compressionBuffer when serializing pages
  - Reuse compression buffer in PagesSerde when serializing pages.
- #13251 Maria Basmanova: Update spatial index memory while building R-Tree
  - Improve memory tracking for geospatial query indexing and prevent worker node down due to out of memory
- #13262 James Sun: Cache row group column stats in ORC column writer
  - Fix high CPU usage when writing ORC files with too many row groups.
- #13310 Rebecca Schlussel: Improve condition for ignoring bucketing for hive
  - Fix computation for number of buckets accessed for `max_buckets_for_grouped_execution`
  - Fix a bug where the bucket column was not available if `max_buckets_for_grouped_execution` was exceeded
- #13312 Wenlei Xie: Add debug support for offline table and partition
  - Add debug mode that allows reading from offline table and partition. This is controlled by session property `offline_data_debug_mode_enabled `
- #13323 Maria Basmanova: Fix ST_Centroid and ST_Buffer for tiny geometries
  - Fix ST_Buffer for tiny geometries (#13194)
  - Fix ST_Centroid for tiny geometries (#10629)
- #13329 Andrii Rosa: Fix grouped execution tagging for table scan
  - Fix bug that causes failure when reading unpartitioned table from some connectors with session property ``grouped_execution_for_eligible_table_scans`` turned on.
- #13334 Andrii Rosa: Add query task count to statistics field of QueryCompletedEvent
  - Add peak running task count to statistics field of QueryCompletedEvent

**Comment by arhimondr** [comment]
```
Hive

* Writing into Hive bucketed tables now done in parallel (multiple threads per node) with respect to the task.writer-count configuration property
```
